### PR TITLE
Make multiple requests to load tickets

### PR DESCRIPTION
**Pull request by bwreid**
This PR separates a single call to the API for all tickets to segmented calls to the API. In total, four calls are made for tickets within 15 days of each other each.

This PR includes pagination.

pagination

This PR will need [this PR from TicketDispenser](https://github.com/WikiEducationFoundation/TicketDispenser/pull/7) to work correctly.

### CODE REVIEW

**Comment by ragesoss** [comment]
eslint error is breaking the build.

**Comment by bwreid** [comment]
Woops. Thought I had fixed that already. New version just pushed up.

File: app/assets/javascripts/components/tickets/tickets_handler.jsx [hunk 1]
```diff
@@ -53,9 +65,12 @@ export class TicketsHandler extends React.Component {
       }
     };
 
+    const TICKETS_PER_PAGE = 10;
+
     const elements = this.props.filteredTickets.map(ticket => (
       <Row key={ticket.id} ticket={ticket} />
-    ));
+    )).slice(this.state.page, this.state.page + TICKETS_PER_PAGE);
+    const pagesLength = Math.floor(elements.length / TICKETS_PER_PAGE) + 1;
```
Comment (ragesoss) [inline app/assets/javascripts/components/tickets/tickets_handler.jsx]: This should be done based on the full filteredTickets array, not just the slice of elements on the current page. I think it also shows an extra page when the ticket count is a multiple of ten. Maybe should be `(count - 1) / TICKETS_PER_PAGE` ?

File: app/assets/javascripts/actions/tickets_actions.js [hunk 2]
```diff
@@ -120,13 +121,40 @@ export const readAllMessages = ticket => async (dispatch) => {
...
+  });
+};
+
+// Fetch as many tickets as possible
 export const fetchTickets = () => async (dispatch) => {
   dispatch({ type: FETCH_TICKETS });
 
-  const response = await fetch('/td/tickets');
-  const json = await response.json();
+  const segments = dateSegments();
```
Comment (ragesoss) [inline app/assets/javascripts/actions/tickets_actions.js]: Locally, I have 107 tickets, I end up with more than 400 in `state.tickets.all`. Also seeing some duplicate key warnings in some cases, so it looks like each ticket is ending up represented multiple times in the set.

Comment (ragesoss) [reply app/assets/javascripts/actions/tickets_actions.js]: Oh, of course, that's because I haven't updated TicketDispenser.

**Comment by ragesoss** [comment]
@bwreid please review the last change I just pushed up. Does the change to the slice params look right?

**Comment by bwreid** [comment]
This looks right to me!

**Comment by ragesoss** [comment]
One thing I ran into before I had the TicketDispenser code updated is duplicate keys, which will throw warnings and potentially cause problems on the frontend. The frontend assumes that each independent request for tickets will return non-overlapping results. Better handling of arbitrary, possibly duplicate sets of tickets may be something we need to handle later on, if we add more/different options for extra queries.

**Pull request merged**